templates/visualbasic/SmartContract1.vb

```diff
@@ -8,9 +10,53 @@ Imports Neo.SmartContract.Framework.Services.Neo
 ' <Assembly: ContractEmail("optional contract email")>
 <Assembly: Features(ContractPropertyState.HasStorage)>
 
-
 Public Class SmartContract1 : Inherits SmartContract
-    Public Shared Sub Main()
-        Storage.Put(Storage.CurrentContext, "Hello", "World")
-    End Sub
+
+    Public Shared Function Main(method As String, args As Object()) As Object
+        If String.Equals(method, "putValue") Then
+            Main = PutValue(args(0))
+        ElseIf String.Equals(method, "getValue") Then
+            Main = GetValue()
+        'ElseIf String.Equals(method, "contractMigrate") Then
+        '    Main = ContractMigrate(args(0), args(1), args(2), args(3), args(4), args(5), args(6), args(7), args(8))
+        Else
+            Main = false
+        End If
+    End Function
+
+    <DisplayName("putValue")>
+    Public Shared Function PutValue(obj As Byte()) As Byte()
+        Storage.Put(Storage.CurrentContext, "StoredData", obj)
+        PutValue = obj
+    End Function
+
+    <DisplayName("getValue")>
+    Public Shared Function GetValue() As Byte()
+        GetValue = Storage.Get(Storage.CurrentContext, "StoredData")
+    End Function
+
+    '<DisplayName("contractMigrate")>
+    'Public Shared Function ContractMigrate(
+    '    script As Byte(), 
+    '    parameterList As Byte(), 
+    '    returnType As Byte, 
```

Comment:
When a contract written in VB.Net has a method with a parameter of a type other than Byte() it causes the Neon compiler to crash.

Example output:

`$ dotnet publish
Microsoft (R) Build Engine version 16.4.0+e901037fe for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 35.71 ms for /Users/david/neo-visual-tracker/sample-workspace/TestVB/TestVB.vbproj.
  TestVB -> /Users/david/neo-visual-tracker/sample-workspace/TestVB/bin/Debug/netstandard2.0/TestVB.dll
  Neo.Compiler.MSIL console app v2.6.0.0
EXEC : LoadModule error : System.NullReferenceException: Object reference not set to an instance of an object. [/Users/david/neo-visual-tracker/sample-workspace/TestVB/TestVB.vbproj]
     at Neo.Compiler.MSIL.OpCode.InitToken(Object _p) in d:\a\1\s\Neo.Compiler.MSIL\MSIL\ILModule.cs:line 795
     at Neo.Compiler.MSIL.ILMethod..ctor(ILType type, MethodDefinition method, ILogger logger) in d:\a\1\s\Neo.Compiler.MSIL\MSIL\ILModule.cs:line 256
     at Neo.Compiler.MSIL.ILType..ctor(ILModule module, TypeDefinition type, ILogger logger) in d:\a\1\s\Neo.Compiler.MSIL\MSIL\ILModule.cs:line 85
     at Neo.Compiler.MSIL.ILModule.LoadModule(Stream dllStream, Stream pdbStream) in d:\a\1\s\Neo.Compiler.MSIL\MSIL\ILModule.cs:line 49
     at Neo.Compiler.Program.Main(String[] args) in d:\a\1\s\Neo.Compiler.MSIL\Program.cs:line 175
/Users/david/neo-visual-tracker/sample-workspace/TestVB/Directory.Build.targets(8,5): error MSB3073: The command "neon TestVB.dll" exited with code 255. [/Users/david/neo-visual-tracker/sample-workspace/TestVB/TestVB.vbproj]`

Reply:
Can you file a bug in the [NEON repo](https://github.com/neo-project/neo-devpack-dotnet) w/ a sample contract demonstrating the issue?

Reply:
> Can you file a bug in the [NEON repo](https://github.com/neo-project/neo-devpack-dotnet) w/ a sample contract demonstrating the issue?

https://github.com/neo-project/neo-devpack-dotnet/issues/197